src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -2081,6 +2246,298 @@ TEST_F(MessageHelperTest, SendGetListOfPermissionsResponse_SUCCESS) {
+      .WillOnce(Return(kCorrelationId));
+  EXPECT_CALL(mock_application_manager_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+  MessageHelper::SendNaviStopStream(kAppId, mock_application_manager_);
+
+  ASSERT_TRUE(result.valid());
+  EXPECT_TRUE((*result)[strings::params].keyExists(strings::function_id));
+  EXPECT_EQ((*result)[strings::params][strings::function_id].asInt(),
+            hmi_apis::FunctionID::Navigation_StopStream);
+  CheckParamsOfMessage(result, true, true);
+}
+
+TEST_F(MessageHelperTest,
+       SendOnAppPermissionsChangedNotification_ExpectSendCorrectMessageToHMI) {
+  std::string policy_app_id("policyappid");
+  policy::AppPermissions dummy_permissions(policy_app_id);
+  dummy_permissions.appRevoked = true;
+  dummy_permissions.isAppPermissionsRevoked = true;
+  std::vector<policy::FunctionalGroupPermission> functional_group_permissions;
+  policy::FunctionalGroupPermission obj;
+  FillFunctionalGroupPermissionObj(obj,
+                                   "group_alias1",
+                                   "group_name1",
+                                   1,
+                                   policy::GroupConsent::kGroupAllowed);
+  functional_group_permissions.push_back(obj);
+  FillFunctionalGroupPermissionObj(obj,
+                                   "group_alias2",
+                                   "group_name2",
+                                   2,
+                                   policy::GroupConsent::kGroupDisallowed);
+  functional_group_permissions.push_back(obj);
+  FillFunctionalGroupPermissionObj(obj,
+                                   "group_alias3",
+                                   "group_name3",
+                                   3,
+                                   policy::GroupConsent::kGroupUndefined);
+  dummy_permissions.appRevokedPermissions = functional_group_permissions;
+  dummy_permissions.appPermissionsConsentNeeded = true;
+  dummy_permissions.appUnauthorized = true;
+  dummy_permissions.priority = "NORMAL";
+  dummy_permissions.requestTypeChanged = true;
+  std::vector<std::string> request_type(3, "dummy");
+  dummy_permissions.requestType = request_type;
+  smart_objects::SmartObjectSPtr result;
+  EXPECT_CALL(mock_application_manager_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+  MessageHelper::SendOnAppPermissionsChangedNotification(
+      kAppId, dummy_permissions, mock_application_manager_);
+  ASSERT_TRUE(result.valid());
+  EXPECT_TRUE((*result)[strings::params].keyExists(strings::function_id));
+  EXPECT_EQ((*result)[strings::params][strings::function_id].asInt(),
+            hmi_apis::FunctionID::SDL_OnAppPermissionChanged);
+  CheckParamsOfMessage(result, true, false);
+  uint16_t number_msg_params = 8;
+  EXPECT_EQ((*result)[strings::msg_params].length(), number_msg_params);
+  EXPECT_TRUE((*result)[strings::msg_params].keyExists("appRevoked"));
```

Comment:
@dtrunov 
I think you can use variables instead of multiple literals here and below